### PR TITLE
docs(migration): add

### PR DESCRIPTION
**Pull request by JacopoDaeli**
Looking for early feedback on how to improve/change this.

### CODE REVIEW

**Comment by JacopoDaeli** [comment]
@decompil3d @jcrugzz what else should we say? Maybe add a bash script that run the webpack config for multi env/locales?

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+```bash
+wrhs upload /my-app/dist example-app --env development --version 1.0.0 --variant en-US
+```
+
+You must build your app for each environment/locale commbination you may need. 
+
+Each build assets must then being properly uploaded to Warehouse using the CLI or the http API directly.
+
+If your app assets are the same across all your enviroments you may choose to build and upload the files only once and simply update the enviroment head when you need. Please note that Warehouse does not rebuild things in your behalf.
```
Comment (decompil3d) [inline MIGRATION.md:54]:
```suggestion
If your app assets are the same across all your environments, you may choose to build and upload the files only once and simply update the environment head when you need. Please note that Warehouse does not rebuild things on your behalf per environment.
```

File: MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+> In the real world, you probably want to automate this using a CICD system such as Jenkins, or Github Actions.
```
Comment (decompil3d) [inline MIGRATION.md:56]: Add a note about minification no longer being automatic based on wrhs.toml -- needs to be manually done in webpack config or as a separate build step.

File: MIGRATION.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  ]
+};
+```
+
+Running `NODE_ENV=development LOCALE=en-US npm run build` will build your app for the `development` environment and `en-US` locale.
+
+The you can use the CLI to upload and register the assets:
+
+```bash
+wrhs upload /my-app/dist example-app --env development --version 1.0.0 --variant en-US
```
Comment (decompil3d) [inline MIGRATION.md:47]: Maybe also show the command to set the head?

**Review by decompil3d — CHANGES_REQUESTED** [review]

File: MIGRATION.md (hunk 3 above)
Comment (JacopoDaeli) [reply MIGRATION.md:71]: Done!

File: MIGRATION.md (hunk 2 above)
Comment (JacopoDaeli) [reply MIGRATION.md:85]: Done this too.

**Comment by JacopoDaeli** [comment]
@decompil3d it should be all set!

File: MIGRATION.md [hunk 4]
```diff
@@ -8,7 +8,19 @@ At the same time we removed building capabilities support. From now on, teams mu
 
 ## No more wrhs.toml
 
-Configuration declared via the `wrhs.toml` is not supported anymore. After moving to the new system, you can safely remove the file.
+Configuration declared via the `wrhs.toml` is not supported anymore. After moving to the new system, you can safely remove the file. If you were turning on the minification capability via the `wrhs.toml`, you now have to implement minification directly in your building pipeline.
+There are many ways to achive this. At GoDaddy we use [terser](https://github.com/terser/terser) webpack plugin whcih can be easily activated via the `webpack.config.js` file of your application:
```
Comment (decompil3d) [inline MIGRATION.md:12]:
```suggestion
There are many ways to achive this. At GoDaddy we use [terser](https://github.com/terser/terser) webpack plugin which can be easily activated via the `webpack.config.js` file of your application:
```

File: MIGRATION.md [hunk 5]
```diff
@@ -8,7 +8,19 @@ At the same time we removed building capabilities support. From now on, teams mu
...
+
+```js
+const TerserPlugin = require('terser-webpack-plugin');
+
+module.exports = {
+  optimization: {
+    minimize: true,
```
Comment (decompil3d) [inline MIGRATION.md:19]: Let's add this comment to the example too

```suggestion
    minimize: true, // or perhaps `process.env.NODE_ENV === 'production'` to minify only in PROD
```

File: MIGRATION.md [hunk 6]
```diff
@@ -41,16 +64,22 @@ module.exports = {
...
 
 ```bash
+# Upload and register v1.0.0 assets
 wrhs upload /my-app/dist example-app --env development --version 1.0.0 --variant en-US
+
+# Set v1.0.0 as current
+wrhs object set-head my-app --env development --version 1.0.0
 ```
 
+> You can defer to updated the head until the new version is ready to be deployed. Each environment has its own head. This means you can upload the assets for all the enviroments at once, and update each individual head whenver the new version is ready to be deployed to that specific environment.
```
Comment (decompil3d) [inline MIGRATION.md:77]:
```suggestion
> You can defer updating the head until the new version is ready to be deployed. Each environment has its own head. This means you can upload the assets for all the enviroments at once, and update each individual head whenver the new version is ready to be deployed to that specific environment.
```

**Review by decompil3d — CHANGES_REQUESTED** [review]

**Pull request merged**